x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java

```diff
@@ -458,6 +465,36 @@ protected static Number count(String index) throws IOException {
         return extractValue(responseAsMap(response), index + ".settings");
     }
 
+    @SuppressWarnings("unchecked")
+    protected static void waitForIdlingSearchableSnapshotsThreadPools() throws Exception {
```

Comment:
I wonder if we need to do it this complicated (although this is a little cleaner in terms of what we assert I guess), wouldn't we get the same effect from just extending the timeout on the busy assert for the cache size above?

Reply:
I'd prefer to not extend timeouts as it does not make obvious why it is needed to wait here. I don't think this change is complicated and it mirrors what we already did at other places.